core/src/test/java/com/oath/oak/OffHeapOakTest.java

```diff
@@ -52,6 +56,7 @@ public void testThreads() throws InterruptedException {
         for (int i = 0; i < NUM_THREADS; i++) {
             threads.get(i).join();
         }
+        assertNull(threadException);
```

Comment:
Threads should be able to get ConcurrentModificationException in this test, since there are concurrent removals, iterations and rebalances.

Reply:
Other exceptions, on the other hand, should not be thrown.
Maybe check `assert threadException == null || threadException isinstanceof ConcurrentModificationException;`.

Reply:
I didn't actually fix the test, just made it so that it won't fail silently. 
@sanastas , if the fix is trivial please let me know and I'll include it

Reply:
Since the fix requires a deeper dive into the test to see which exceptions are expected and which aren't, I'm leaving issue #52 open until the test and/or relevant code is actually fixed